mantidproject/mantid
# Fix Crash When Running Multiple Reductions in the Preview Tab

**Description of work.**
Add some notifying calls between the job runner and the preview view to disable the tab while the reduction is running, which was causing some race-based segfaults if the load button was spammed. 

The cause of the segfaults appeared to be when two reductions were started very close together. When the second one finished, the other tried to access a validator that the first one had already cleared, so was trying to deref null pointers. 

**To test:**
1. Open the ISIS Reflectometry interface
2. Go to the Preview tab
3. Enter run `INTER45455_inst` and click Load (This data is available on this Epic and must be in your user search path: #31069)
4. Put an angle of 0 and try to click load a bunch. The button should be disabled while processing. The algo should error, but not crash.
5. Now put an angle in and click load a lot again. The line plot should be filled in and it shouldn't crash. 
6. Now draw a square on the inst viewer and move your drawn area as many times as you can in the shortest time to try and force reductions to overlap. The tab should disable itself while the reduction is running and prohibit you from doing this. 

Fixes #34865 

*This does not require release notes* because **this tab isn't user facing yet.**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
